### ISSUE TITLE
Block checksum verification is called while verify_checksums is set to false

### ISSUE THREAD

**Issue by ahamlat** [facebook/rocksdb#11002]
Hi team,

When using rocksdbjni Java library, version 7.7.3, we've noticed that checksum verification consumes more than 50% in CPU time during the execution of RocksDB.get method. My understanding is that this happens when we have a cache miss in block cache. Our block cache is configured to 1 GiB as you can see in the LOG file screenshot below (2nd one).

<img width="1724" alt="image" src="https://user-images.githubusercontent.com/5099602/204511641-86ab4b9b-45e5-4a2b-a13d-81fa26435d38.png">

I wanted to disable checksum verification but it didn't work :

```
private final ReadOptions readOptions = new ReadOptions().setVerifyChecksums(false);

...

@Override
  public Optional<byte[]> get(final RocksDbSegmentIdentifier segment, final byte[] key)
      throws StorageException {
    throwIfClosed();

    try (final OperationTimer.TimingContext ignored = metrics.getReadLatency().startTimer()) {
      return Optional.ofNullable(db.get(segment.get(), readOptions, key));
    } catch (final RocksDBException e) {
      throw new StorageException(e);
    }
  }
```
This case is very similar to this issue[ #9718](https://github.com/facebook/rocksdb/issues/9718)  fixed in this PR [#9767](https://github.com/facebook/rocksdb/pull/9767)

I don't understand why verifyChecksum is still called, my understanding of the code [here](https://github.com/facebook/rocksdb/blob/727bad78b8295bc43674a4087df0f7ce9a69ee2b/table/block_fetcher.cc#L35) is that VerifyBlockChecksum shouldn't be called if verify_checksums is set to false in read_options_. Am I missing something (I'm not a C++ expert) ?

I've also tried to set checksumType to ChecksumType.kNoChecksum in BlockBasedTableConfig instance, which set checksum to 0 in database/LOG file, but my understanding [here](https://github.com/facebook/rocksdb/blob/8ae0c33a7a5758a282a14a7bc0a76aa4d6017587/java/src/main/java/org/rocksdb/ChecksumType.java#L13) is that kNoChecksum is not yet implemented.

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/5099602/204515661-8feaa216-43b3-4c77-a3cc-56efbb992e31.png">

I don't know if it is relevant for this case but I have enabled Bloom filters.

### Expected behavior
VerifyBlockChecksum method shouldn't be called when verify_checksums is set to false in read_options.

### Actual behavior
VerifyBlockChecksum method is called when verify_checksums is set to false in read_options.

### Steps to reproduce the behavior
Call RocksDB.get Java method with columnFamilyHandle, read_options with verify_checksums set to false and the key.

```public byte[] get(final ColumnFamilyHandle columnFamilyHandle, final ReadOptions opt, final byte[] key)```

**Comment by pdillinger** [comment]
`ReadOptions::verify_checksums` is not a well-defined feature, unfortunately. If I do Get("a") with verify_checksums=false and that loads a block into block cache, and later do Get("b") with verify_checksums=true, is the correct behavior to use the block from block cache that hasn't had its checksum verified (compromised data integrity) or to only load blocks into block cache if their checksums are verified (unnecessary I/O)? (We don't store block checksums in block cache.) Arguably, unrequested checksum calculation is a lesser of evils compared to compromised data integrity or extra I/O.

**Comment by ajkr** [comment]
By the way, #10427 began fixing this. IIRC it appeared not far from completion during my last review. If you want to pick up that work I would be glad to review it.

**Comment by ahamlat** [comment]
Thanks @pdillinger and @ajkr. Regarding data integrity, we already do data integrity controls at the application level. We're building an Ethereum client, the data we fetch from the database  belongs either to blocks or to the World State. We're running validations on block headers and bodies, but also on the world state by calculating the state root hash of the Patricia Merkle Trie data structure. This will validate that data integrity is not compromised.
Just curious, what should happen if checksum verification returns false (checksum doesn't match) in the case we enable checksum verification ?

**Comment by ahamlat** [comment]
Fixed in RocksDB 8.0.0
